include/flatbuffers/stl_emulation.h

```diff
@@ -28,14 +28,16 @@
 
 // Detect C++17 compatible compiler.
 // __cplusplus >= 201703L - a compiler has support of 'static inline' variables.
-#if defined(FLATBUFFERS_USE_STD_OPTIONAL) \
-    || (defined(__cplusplus) && __cplusplus >= 201703L) \
-    || (defined(_MSVC_LANG) &&  (_MSVC_LANG >= 201703L))
-  #include <optional>
-  #ifndef FLATBUFFERS_USE_STD_OPTIONAL
-    #define FLATBUFFERS_USE_STD_OPTIONAL
-  #endif
-#endif // defined(FLATBUFFERS_USE_STD_OPTIONAL) ...
+#if !defined(FLATBUFFERS_DONT_USE_STD_OPTIONAL)
+  #if defined(FLATBUFFERS_USE_STD_OPTIONAL) \
+      || (defined(__cplusplus) && __cplusplus >= 201703L) \
```

Comment:
Why not just make these && with FLATBUFFERS_USE_STD_OPTIONAL? It looks like FLATBUFFERS_USE_STD_OPTIONAL is opt-in anyways, as we don't explicitly set it anywhere.

Reply:
Sure, that sounds fine.

Reply:
Sorry, i meant to not have FLATBUFFERS_DONT_USE_STD_OPTIONAL at all, and just use something like 

```
defined(FLATBUFFERS_USE_STD_OPTIONAL) && (\
    || (defined(__cplusplus) && __cplusplus >= 201703L) \
    || (defined(_MSVC_LANG) &&  (_MSVC_LANG >= 201703L)))
```

Reply:
Ah, I see - have it default to not using `std::optional`. Yeah, that works too.

Reply:
Thanks!